wrappers/python/CMakeLists.txt

```diff
@@ -14,50 +14,23 @@ if (NOT hasParent)
     option (BUILD_SHARED_LIBS "Link python module to shared lib" OFF)
     option (BUILD_WRITERS "Build with writer support (encoders)" ON)
     option (BUILD_READERS "Build with reader support (decoders)" ON)
-
-    # build the main library
-    if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../../core)
-        # In development mode, when the whole zxing-cpp directory is checked out, build against head code.
-        add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../../core ZXing EXCLUDE_FROM_ALL)
-
-        include(${CMAKE_CURRENT_SOURCE_DIR}/../../zxing.cmake)
-        zxing_add_package(pybind11 pybind11 ${pybind11_git_repo} ${pybind11_git_rev})
+    if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/core)
+        add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/core ZXing EXCLUDE_FROM_ALL)
+        include(${CMAKE_CURRENT_SOURCE_DIR}/zxing.cmake)
     else()
-        # we don't have access to the top-level cmake helpers -> simply fetch it unconditional
-        include(FetchContent)
-        FetchContent_Declare (pybind11
-            GIT_REPOSITORY ${pybind11_git_repo}
-            GIT_TAG ${pybind11_git_rev})
-        FetchContent_MakeAvailable (pybind11)
-
-        # Building from python source distribution (which does not include the whole repository but only python part)
-        # so we need to get c++ source git to build the python extension. The python distribution version (given in
-        # VERSION_INFO), matches the c++ version the distribution must build against ; hence the checkout of
-        # related tag.
-        find_package (Git REQUIRED)
-        execute_process(
-            COMMAND ${GIT_EXECUTABLE} ls-remote https://github.com/zxing-cpp/zxing-cpp.git tags/*
-            RESULT_VARIABLE RESULT
-            OUTPUT_VARIABLE OUTPUT)
-        string (FIND "${OUTPUT}" "refs/tags/v${VERSION_INFO}\n" FOUND_IDX)
-        if (FOUND_IDX LESS 0)
-            # Version not found in tags (suppose we are preparing future version), use master branch
-            FetchContent_Declare(zxing-cpp
-                GIT_REPOSITORY https://github.com/zxing-cpp/zxing-cpp.git)
-        else()
-            FetchContent_Declare(zxing-cpp
-                GIT_REPOSITORY https://github.com/zxing-cpp/zxing-cpp.git
-                GIT_TAG v${VERSION_INFO})
-        endif()
-        if(NOT zxing-cpp_POPULATED)
-            FetchContent_Populate(zxing-cpp)
-            add_subdirectory(${zxing-cpp_SOURCE_DIR}/core ZXing EXCLUDE_FROM_ALL)
-        endif()
+        message(FATAL_ERROR "Unable to locate zxing source code")
     endif()
 else()
     zxing_add_package(pybind11 pybind11 ${pybind11_git_repo} ${pybind11_git_rev})
 endif()
 
+# Check pybind helpers are availible
+find_package(pybind11 CONFIG)
```

Comment:
`find_package` is automatically called inside `zxing_add_package` if `BUILD_DEPENDENCIES` is set to `AUTO` which happens in the toplevel `CMakeLists.txt` which is only available/evaluated if `hasParent` is true. So in the `sdist` case, the following line should do the trick if added near line 16 above: `set(BUILD_DEPENDENCIES "AUTO")`.

With that in-place one single `zxing_add_package` call and no extra `find_package` should hopefully work for all use cases. Could you please check if I'm correct with this assessment?

Reply:
I'll have to have a look.  I recall this was a workaround for a Windows build failure, when it couldn't locate pybind.  Will check if your suggestion helps

Reply:
Worked great with that change across all the platforms.  There's a single `zxing_add_package ` now for finding pybind11